examples/cluster/plot_cluster_comparison.py

```diff
@@ -3,116 +3,168 @@
 Comparing different clustering algorithms on toy datasets
 =========================================================
 
-This example aims at showing characteristics of different
+This example aims to show characteristics of different
 clustering algorithms on datasets that are "interesting"
-but still in 2D. The last dataset is an example of a 'null'
-situation for clustering: the data is homogeneous, and
-there is no good clustering.
-
-While these examples give some intuition about the algorithms,
-this intuition might not apply to very high dimensional data.
-
-The results could be improved by tweaking the parameters for
-each clustering strategy, for instance setting the number of
-clusters for the methods that needs this parameter
-specified. Note that affinity propagation has a tendency to
-create many clusters. Thus in this example its two parameters
-(damping and per-point preference) were set to mitigate this
-behavior.
+but still in 2D. With the expection of the last dataset,
+the parameters of each of these dataset-algorithm pairs
+has been tuned to produce good clustering results. Some
+algorithms are more sensitive to parameter values than
+others.
+
+The last dataset is an example of a 'null' situation for
+clustering: the data is homogeneous, and there is no good
+clustering. For this example, the null dataset uses the
+same parameters as the dataset in the row above it, which
+represents a mismatch in the parameter values and the
+data structure.
+
+While these examples give some intuition about the
+algorithms, this intuition might not apply to very high
+dimensional data.
 """
 print(__doc__)
 
 import time
+import warnings
 
 import numpy as np
 import matplotlib.pyplot as plt
 
-from sklearn import cluster, datasets
+from sklearn import cluster, datasets, mixture
 from sklearn.neighbors import kneighbors_graph
 from sklearn.preprocessing import StandardScaler
 
 np.random.seed(0)
 
+colors = np.array([x for x in 'bgrcmykbgrcmykbgrcmykbgrcmyk'])
+colors = np.hstack([colors] * 20)
```

Comment:
I don't like this.
Create color as `['r', 'g', ... 'b']` and then use the modulo during the plot.